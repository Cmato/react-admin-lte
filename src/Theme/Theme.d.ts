import * as React from 'react';

export interface ThemeProps {
  boxed?: boolean;
  defaultSidebarCollapsed?: boolean;
  fixed?: boolean;
  miniSidebar?: boolean;
  onChange?: (collapsed: boolean, event: React.ChangeEvent<{}>) => void,
  sidebarCollapsed?: boolean;
  skinColor?: SkinColor;
  topNav?: boolean;
}

export type SkinColor = 'blue' | 'black' | 'purple' | 'green' | 'red' |
  'yellow' | 'blue-light' | 'black-light' | 'purple-light' | 'green-light' |
  'red-light' | 'yellow-light';

declare const Theme: React.ComponentType<ThemeProps>;

export default Theme;
