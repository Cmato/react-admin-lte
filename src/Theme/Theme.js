import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import warning from 'warning';
import uncontrollable from 'uncontrollable';

export class Theme extends Component {
  constructor(props) {
    super(props);
    this.handleChange= this.handleChange.bind(this);
  }

  getChildContext() {
    return {
      $adminlte_theme: {
        onChange: this.handleChange
      }
    }
  }

  handleChange(e) {
    this.props.onChange(!this.props.sidebarCollapsed, e);
  }

  render() {
    const {
      boxed,
      children,
      className: classNameProp,
      fixed,
      miniSidebar,
      onChange: _,
      sidebarCollapsed,
      skinColor,
      topNav,
      ...other
    } = this.props;

    warning(
      !(boxed === true && fixed == true),
      'The boxed and fixed properties should not be supplied at the same time.'
    );

    const className = classNames({
      fixed,
      'layout-boxed': boxed,
      'layout-top-nav': topNav,
      'sidebar-collapse': sidebarCollapsed,
      'sidebar-mini': miniSidebar,
      [`skin-${skinColor}`]: skinColor
    }, classNameProp);

    return (
      <div className={className} {...other}>
        <div className="wrapper">
          {children}
        </div>
      </div>
    );
  }
}

Theme.propTypes = {
  /**
   * Layout will only be stretched to 1250px.
   */
  boxed: mutuallyExclusiveTrueProps('boxed', 'fixed'),
  /**
   * The content of the layout.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * It `true`, collapsed the sidebar by default.
   */
  defaultSidebarCollapsed: PropTypes.bool,
  /**
   * Displays fixed header and sidebar.
   */
  fixed: mutuallyExclusiveTrueProps('boxed', 'fixed'),
  /**
   * When collapsed, sidebar will show icons only.
   */
  miniSidebar: PropTypes.bool,
  /**
   * Callback fired when the sidebar is toggled.
   *
   * @param {object} event The event source of the callback
   * @param {boolean} collapsed The 'collapsed` state of the sidebar
   */
  onChange: PropTypes.func,
  /**
   * If `true`, collapses the sidebar. Setting this prop enables control of
   * the Layout.
   */
  sidebarCollapsed: PropTypes.bool,
  /**
   * The color of the skin.
   */
  skinColor: PropTypes.oneOf([
    'blue', 'black', 'purple', 'green', 'red', 'yellow', 'blue-light', 'black-light',
    'purple-light', 'green-light', 'red-light', 'yellow-light'
  ]),
  /**
   * Removes sidebar and puts navbar at top.
   */
  topNav: PropTypes.bool
};

Theme.childContextTypes = {
  $adminlte_theme: PropTypes.shape({
    onChange: PropTypes.func
  })
};

export default uncontrollable(Theme, {
  sidebarCollapsed: 'onChange'
});
