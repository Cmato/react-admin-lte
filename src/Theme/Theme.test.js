import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { mount, shallow } from 'enzyme';
import { Theme } from './Theme';

it('renders empty', () => {
  expect(shallow((
    <Theme>
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders fixed', () => {
  expect(shallow((
    <Theme fixed={true}>
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders boxed', () => {
  expect(shallow((
    <Theme boxed={true}>
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders top navigation', () => {
  expect(shallow((
    <Theme topNav={true}>
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders mini sidebar', () => {
  expect(shallow((
    <Theme miniSidebar={true}>
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders collapsed sidebar', () => {
  expect(shallow((
    <Theme
      sidebarCollapsed={true}
      onChange={() => { return; }}
    >
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders skin', () => {
  expect(shallow((
    <Theme skinColor="blue">
      <div />
    </Theme>
  ))).toMatchSnapshot();
});

it('renders changed collapsed sidebar on toggle', () => {
  const sentEvent = {};
  let receivedCollapsed = false;
  let receivedEvent;

  class SidebarToggler extends Component {
    static contextTypes = {
      $adminlte_theme: PropTypes.shape({
        onChange: PropTypes.func
      })
    };

    componentDidMount() {
      this.context.$adminlte_theme.onChange(sentEvent);
    }

    render() {
      return null;
    }
  }

  mount(
    <Theme onChange={(collapsed, e) => {
      receivedCollapsed = collapsed;
      receivedEvent = e;
    }}>
      <SidebarToggler />
    </Theme>
  );

  expect(receivedCollapsed).toEqual(true);
  expect(receivedEvent).toEqual(sentEvent);
});
