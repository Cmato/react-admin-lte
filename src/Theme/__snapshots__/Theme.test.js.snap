// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders boxed 1`] = `
<div
  className="layout-boxed"
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;

exports[`renders collapsed sidebar 1`] = `
<div
  className="sidebar-collapse"
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;

exports[`renders empty 1`] = `
<div
  className=""
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;

exports[`renders fixed 1`] = `
<div
  className="fixed"
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;

exports[`renders mini sidebar 1`] = `
<div
  className="sidebar-mini"
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;

exports[`renders skin 1`] = `
<div
  className="skin-blue"
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;

exports[`renders top navigation 1`] = `
<div
  className="layout-top-nav"
>
  <div
    className="wrapper"
  >
    <div />
  </div>
</div>
`;
